source/braille.py

```diff
@@ -1696,6 +1703,19 @@ def setDisplayByName(self, name, isFallback=False, detected=None):
 			self.setDisplayByName("noBraille", isFallback=True)
 			return False
 
+	def _onBrailleViewerChangedState(self, created):
+		import brailleViewer
+		if not created:  # the viewer was destroyed
+			# if we have a hardware display attached, use it's cell count
+			self.displaySize = None if not self.display else self.display.numCells

```

Comment:
I don't think displaySize should be an optional:

```suggestion
			self.displaySize = 0 if not self.display else self.display.numCells
```